snuba/web/split.py

```diff
@@ -328,6 +328,10 @@ def execute(
             metrics.increment("column_splitter.query_above_limit")
             return None
 
+        # Do not split if there is already a condition on an ID column
+        if self.__id_column in query.get_columns_referenced_in_conditions():
+            return None
```

Comment:
This is only applying the optimization on the legacy query representation, which is not used anymore. The code would be removed starting end of this week.

Reply:
Ops, thanks. Updated to use the AST.